### PR TITLE
Compile regexp only once and use it as many times wanted.

### PR DESCRIPTION
**Pull request by balamurugana**

### CODE REVIEW

File: pkg/s3signer/utils.go [hunk 1]
```diff
@@ -77,6 +77,9 @@ func queryEncode(v url.Values) string {
 	return buf.String()
 }
 
+// if object matches reserved string, no need to encode them
+var ReservedObjectNames = regexp.MustCompile("^[a-zA-Z0-9-_.~/]+$")
```
Comment (harshavardhana) [inline pkg/s3signer/utils.go]: no need to export this..

File: pkg/s3signer/utils.go [hunk 2]
```diff
@@ -85,9 +88,7 @@ func queryEncode(v url.Values) string {
 // This function on the other hand is a direct replacement for url.Encode() technique to support
 // pretty much every UTF-8 character.
 func urlEncodePath(pathName string) string {
```
Comment (harshavardhana) [inline pkg/s3signer/utils.go]: In-fact this function should be exposed..

File: api.go [hunk 3]
```diff
@@ -278,6 +278,9 @@ type requestMetadata struct {
 	contentMD5Bytes    []byte
 }
 
+var regCred = regexp.MustCompile("Credential=([A-Z0-9]+)/")
```
Comment (harshavardhana) [inline api.go]: Add comments for these regexes.

Comment (balamurugana) [inline api.go]: Done

**Pull request merged**